### ISSUE TITLE
Handle smaller batch sizes for ResNet example

### ISSUE THREAD

**Issue by michaelklachko** [ppwwyyxx/tensorpack#655]
In ResNet example, the expected value for  --batch argument is 256, or multiple of 256. That's required to reproduce claimed accuracy. However, the model should still work with smaller batch sizes (with potentially degraded accuracy). 

Currently, with --batch = X, where X < 256, learning rate will be set to zero after 3 epochs of training:
https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/imagenet-resnet.py#L76
This becomes an issue when we need to reduce batch size to fit larger models in GPU memory.

I suggest replacing the int division with floating point, and (optionally) printing a warning about potential accuracy degradation, either once in the beginning, or next to val-error results, perhaps something like this:

```
if args.batch < 256:
    tag = ' (accuracy might be degraded because batch size < 256)'
else:
    tag = ''
infs = [ClassificationError('wrong-top1', 'val-error-top1'+tag),
        ClassificationError('wrong-top5', 'val-error-top5'+tag)]
```
